### PR TITLE
Replaced sendgrid with sparkpost

### PR DESCRIPTION
**Pull request by agnessa**
I had to:
- use a fork with a version of sparkpost_rails compatible with Rails 7
- use the `html_content_only=true` setting in the sparkpost initialiser, as otherwise the devise emails would come as plain text with html tags

## Testing instructions

Can be tested in development, API key needed in .env

## Tracking

https://vizzuality.atlassian.net/browse/RA2-156?atlOrigin=eyJpIjoiMGViOThmODUyNDhjNGMwNmE2OTM5NTNlNThkMDNjZTAiLCJwIjoiaiJ9

### CODE REVIEW

File: backend/config/environments/development.rb [hunk 1]
```diff
@@ -40,17 +40,7 @@
...
-  ActionMailer::Base.smtp_settings = {
-    address: "smtp.sendgrid.net",
-    port: "587",
-    authentication: :plain,
-    user_name: ENV["SENDGRID_USERNAME"],
-    password: ENV["SENDGRID_PASSWORD"],
-    domain: "localhost",
-    enable_starttls_auto: true
-  }
+  config.action_mailer.delivery_method = :sparkpost
```
Comment (martintomas) [inline backend/config/environments/development.rb:43]: Do you think it is good idea to use for example `letter_opener` and not use smtp server at develop?

**Review by martintomas — APPROVED** [review]
I am now not sure, but I have feeling that link (url) at forgotten password email was wrong (it used Prod server instead of Stage). Maybe it is not part of this PR, but could you check that this email and link inside are okay? If you don't have time for it, just let me know and I will do it, no problem ;)

File: backend/config/environments/development.rb (hunk 1 above)
Comment (agnessa) [reply backend/config/environments/development.rb:43]: OK, agree!

**Comment by martintomas** [comment]
> > 
> 
> indeed, since both production and staging servers run in production environment, the mailer host is going to be the same :/ INSTANCE_ROLE it is

I have added `BACKEND_URL` variable some time ago, isn't this something which we can use to set up mailer host?

**Comment by agnessa** [comment]
> > > 
> > 
> > 
> > indeed, since both production and staging servers run in production environment, the mailer host is going to be the same :/ INSTANCE_ROLE it is
> 
> I have added `BACKEND_URL` variable some time ago, isn't this something which we can use to set up mailer host?

good point, I did that (and also added these new env vars in prod)

**Pull request merged**